src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -241,4 +230,116 @@ private void emitGlobalSetter(Constant segmentConstant, Constant vhConstant, Str
             """);
         decrAlign();
     }
+
+    public String emitGlobalSegment(MemoryLayout layout, String javaName, String nativeName, Declaration declaration) {
+        String mangledName = mangleName(javaName, MemorySegment.class);
+        incrAlign();
+        appendLines(STR."""
+            private static final MemorySegment \{mangledName} = RuntimeHelper.lookupGlobalVariable(\"\{nativeName}\", \{layoutString(0, layout)});
```

Comment:
Shouldn't this re-use the layout field we emit instead of emitting the same layout string again inline?

Reply:
good point